mne/stats/tests/test_cluster_level.py

```diff
@@ -911,121 +911,77 @@ def test_new_cluster_api(Inst):
     """Test handling different MNE objects in the cluster API."""
     pd = pytest.importorskip("pandas")
 
-    n_subs, n_epo, n_chan, n_freq, n_times = 2, 2, 3, 4, 5
-    info = create_info(ch_names=n_chan, sfreq=1000, ch_types="eeg")
-    # Introduce a significant difference in a specific region, time, and frequency
-    region_start = 1
-    region_end = 2
-    time_start = 2
-    time_end = 4
-    freq_start = 2
-    freq_end = 4
-
-    if Inst == EpochsArray:
-        # Create random data for EpochsArray
-        inst1 = Inst(np.random.randn(n_epo, n_chan, n_times), info=info)
-        # Adding a constant to create a difference
-        data_copy = inst1.get_data().copy()  # no data attribute for EpochsArray
-        data_copy[:, region_start:region_end, time_start:time_end] += (
-            2  # Modify the copy
-        )
-        inst2 = Inst(
-            data=data_copy, info=info
-        )  # Use the modified copy as a new instance
-
-    elif Inst == EvokedArray:
-        # Create random data for EvokedArray
-        inst1 = Inst(np.random.randn(n_chan, n_times), info=info)
-        data_copy = inst1.data.copy()
-        data_copy[region_start:region_end, time_start:time_end] += 2
-        inst2 = Inst(data=data_copy, info=info)
-
-    elif Inst == EpochsTFRArray:
-        # Create random data for EpochsTFRArray
-        data_tfr1 = np.random.randn(n_epo, n_chan, n_freq, n_times)
-        data_tfr2 = np.random.randn(n_epo, n_chan, n_freq, n_times)
-        inst1 = Inst(
-            data=data_tfr1, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
-        inst2 = Inst(
-            data=data_tfr2, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
-        data_tfr2 = inst2.data.copy()
-        data_tfr2[
-            :, region_start:region_end, freq_start:freq_end, time_start:time_end
-        ] += 2
-        inst2 = Inst(
-            data=data_tfr2, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
+    rng = np.random.default_rng(seed=8675309)
```

Comment:
we should always explicitly set a random seed when testing. Otherwise there's a (slim) chance the test might fail occasionally just from randomness. Here, I create an `np.random.Generator` object called `rng` and then later call `rng.normal()` to actually generate the random values.